File: app4web_05042021/app/src/main/java/site/app4web/app4web/Core/JasonCallback.java
package site.app4web.app4web.Core;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

import site.app4web.app4web.Helper.JasonHelper;

import org.json.JSONObject;

public class JasonCallback {
    public void href(Intent intent, final JSONObject options) {
        try {
            JSONObject action = options.getJSONObject("action");
            JSONObject event = options.getJSONObject("event");
            Context context = (Context)options.get("context");

            String return_string = intent.getStringExtra("return");
            JSONObject return_value = new JSONObject(return_string);
            JasonHelper.next("success", action, return_value, event, context);

        } catch (Exception e) {
            Log.d("Warning", e.getStackTrace()[0].getMethodName() + " : " + e.toString());
        }
    }
}

File: app4web_k9/app/src/main/java/site/app4web/app4web/Core/JasonCallback.kt
package site.app4web.app4web.Core

import android.content.Context
import android.content.Intent
import android.util.Log
import org.json.JSONObject
import site.app4web.app4web.Helper.JasonHelper

class JasonCallback {
    fun href(intent: Intent, options: JSONObject) {
        try {
            val action = options.getJSONObject("action")
            val event = options.getJSONObject("event")
            val context = options["context"] as Context
            val return_string = intent.getStringExtra("return")
            val return_value = JSONObject(return_string!!)
            JasonHelper.next("success", action, return_value, event, context)
        } catch (e: Exception) {
            Log.d(
                "Warning",
                e.stackTrace[0].methodName + " : " + e.toString()
            )
        }
    }
}